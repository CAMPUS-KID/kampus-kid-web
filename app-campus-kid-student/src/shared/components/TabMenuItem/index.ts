export { default as TabMenuItem } from './TabMenuItem';
