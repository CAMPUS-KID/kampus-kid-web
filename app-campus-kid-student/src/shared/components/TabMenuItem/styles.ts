import { css } from '@emotion/native';
import { Theme } from '@theme';

export const styleSheet = ({ fontFamilies, fontSizes }: Theme) => ({
  item: css`
    flex: 1;
    align-items: center;
    padding: 10px 0px;
  `,
  text: css`
    font-family: ${fontFamilies.inter};
    font-size: ${fontSizes.mini};
  `
});
