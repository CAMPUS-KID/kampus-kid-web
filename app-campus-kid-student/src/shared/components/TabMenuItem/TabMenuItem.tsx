import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { useTheme } from '@emotion/react';

import { useBuildStyles, useTesting } from '@shared_hooks';
import { Components } from '@shared_constants';
import { Theme } from '@theme';

import { styleSheet } from './styles';
import { useNavigate } from '@shared_hooks';
import { IconEnum } from '@shared_enums';
import { Icon } from '../Icon';

export interface Props {
  route: string;
  icon: IconEnum;
  tabName: string;
  isActive: boolean;
}

const TabMenuItem = ({ route, icon, tabName, isActive }: Props) => {
  const { getTestingProps } = useTesting(Components.TAB_BAR_ITEM);
  const styles = useBuildStyles(styleSheet);
  const theme = useTheme() as Theme;
  const navigate = useNavigate();

  const color = isActive ? theme.colors.accent : theme.colors.tertiary;

  return (
    <TouchableOpacity
      {...getTestingProps(`tab-item`)}
      style={styles.item}
      onPress={() => navigate(route)}
    >
      <Icon name={icon} color={color} fill={color} width={30} height={30} />
      <Text style={[styles.text, { color }]}>{tabName}</Text>
    </TouchableOpacity>
  );
};
export default TabMenuItem;
