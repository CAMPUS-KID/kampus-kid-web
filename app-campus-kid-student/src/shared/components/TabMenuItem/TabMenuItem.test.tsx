import React from 'react';
import TabMenuItem, { Props } from './TabMenuItem';
import { render } from 'react-native-testing-library';
import { IconEnum } from '@shared_enums';

describe('TabBar', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      route: 'Home',
      tabName: 'Home',
      isActive: true,
      icon: IconEnum.MOVIES
    };
  });

  it('should render', () => {
    const { toJSON } = render(<TabMenuItem {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
