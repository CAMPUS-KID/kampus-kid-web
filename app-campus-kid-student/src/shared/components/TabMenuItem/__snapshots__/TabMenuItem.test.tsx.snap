// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar should render 1`] = `
<View
  accessibilityLabel="tab item"
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 10,
    }
  }
  testID="tab-bar-item-tab-item"
>
  <SvgImage
    color="#F00303"
    fill="#F00303"
    height={30}
    width={30}
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Inter",
          "fontSize": 10,
        },
        Object {
          "color": "#F00303",
        },
      ]
    }
  >
    Home
  </Text>
</View>
`;
